perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,288 @@
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_REGEX = re.compile(r'^;;; \s+ (.*)', re.X | re.M)
+
+  SCORE_REGEX = re.compile(r'''
+    ^ (Composite \s+ Score) : \s+ (\d+ \. \d+)
+  ''', re.X | re.M)
+  RESULT_REGEX_C = re.compile(r'''
+    ^
+    ( .+? ) \s+  #1: Test name
+    Mflops: \s+
+    ( \d+ \. \d+ )  #2: Test score
+    ( \s+ \( .+? \) )?  #3: Optional test details
+  ''', re.X | re.M)
+  RESULT_REGEX_JAVA = re.compile(r'''
+    ^
+    ( .+? )  #1: Test name
+    : \s+
+    ( \d+ \. \d+ )  #2: Test score
+  ''', re.X | re.M)
+  PLATFORM_REGEX = re.compile(r'''
+    ^
+    ( \w+ \. \w+ )  #1: Property name
+    : \s+
+    ( .* )  #2: Property value
+  ''', re.X | re.M)
+
+  def FindBenchStart(results, start_index=0):
+    m = RESULT_START_REGEX.search(results, start_index)
+    if m is None:
+      return -1, 'Unknown'
+    return m.start(), m.group(1)
+
+  def ExtractPlatform(result, bench_version):
+    metadata = {}
+    meta_start = None
+    if bench_version.startswith('C'):
+      pass
+    elif bench_version.startswith('Java'):
+      for m in PLATFORM_REGEX.finditer(result):
+        if meta_start is None:
+          meta_start = m.start()
+        metadata[m.group(1)] = m.group(2)
+    return metadata, meta_start
+
+  def ExtractScore(result, bench_version):
+    m = SCORE_REGEX.search(result)
+    label = m.group(1)
+    score = float(m.group(2))
+    return score, label, m.end()
+
+  def ExtractResults(result, bench_version):
+    datapoints = []
+    if bench_version.startswith('C'):
+      for groups in regex_util.ExtractAllMatches(RESULT_REGEX_C, result):
+        metric = '{0} {1}'.format(groups[0].strip(), groups[2].strip())
+        metric = metric.strip().strip(':')  # Extra ':' in 'MonteCarlo:'.
+        value = float(groups[1])
+        datapoints.append((metric, value))
+    elif bench_version.startswith('Java'):
+      for groups in regex_util.ExtractAllMatches(RESULT_REGEX_JAVA, result):
+        datapoints.append((groups[0].strip(), float(groups[1])))
+    return datapoints
+
+  # Find start positions for all the test results.
+  tests = []
+  test_start_pos = 0
+  while True:
+    start_index, bench_version = FindBenchStart(results, test_start_pos)
+    if start_index == -1:
+      break
+    tests.append((start_index, bench_version))
+    test_start_pos = start_index + 1
+
+  # Now loop over individual tests collecting samples.
+  samples = []
+  for test_num, (start_index, bench_version) in enumerate(tests):
+    # Get end index - either start of next test, or None for the last test.
+    end_index = None
+    if test_num + 1 < len(tests):
+      end_index = tests[test_num + 1][0]
+    result = results[start_index:end_index]
+
+    metadata = {'bench_version': bench_version}
```

Comment:
Also, would it be possible to report the benchmark language and size as separate metadata items?